src/Examples/ExampleMemoryMappedFile.cpp

```diff
@@ -31,10 +31,22 @@ MemoryMappedFile::Handle MemoryMappedFile::Open(const char* path)
 		CloseHandle(fileMapping);
 		CloseHandle(file);
 
-		return Handle { INVALID_HANDLE_VALUE, INVALID_HANDLE_VALUE, nullptr };
+		return Handle { INVALID_HANDLE_VALUE, INVALID_HANDLE_VALUE, nullptr, 0 };
 	}
 
-	return Handle { file, fileMapping, baseAddress };
+	MEMORY_BASIC_INFORMATION memoryInfo;

```

Comment:
VirtualQuery will give you the size of the mapping into the address space, which is not necessarily the same as the file size on disk.
You should be able to use GetFileInformationByHandle() in this case instead.

Reply:
> VirtualQuery will give you the size of the mapping into the address space, which is not necessarily the same as the file size on disk. You should be able to use GetFileInformationByHandle() in this case instead.

I think that the size of the mapped bytes in the address space is exactly what I need here because it must be related to the pointer passed into `PDB::ValidateFile` and `PDB::RawFile`. If the mapped byte size is less than the file size, it should be detected in `PDB::ValidateFile` to avoid an Access Violation.

Reply:
Since we explicitly map the entire file, the size of the memory-mapped file can never be smaller than the actual size on disk. In fact, it's the opposite.

Windows (and other OSs) map files in page granularity, which is 4kB on Windows. This means that the size of the mapping will always be a multiple of the page size, so it's not suitable for performing the actual size checks you implemented.

Take the absolute worst case: a file that consists of a single byte.
The size of the memory mapped file will be 4096 and you would pass that to the validation function, where none of the size checks would fail, even though there's only one byte stored in the file.

Down the line this wouldn't lead to an access violation, but even worse: the code thinks it's reading valid data, but the data is not actually there -- just zeroes mapped into the address space.

Reply:
I see, thank you for explaining. Fixed.